ambry-quota/src/main/java/com/github/ambry/quota/capacityunit/AmbryCUQuotaEnforcer.java

```diff
@@ -118,11 +129,20 @@ public void shutdown() {
    * Build the {@link QuotaRecommendation} object from the specified usage and {@link QuotaName}.
    * @param usage percentage usage.
    * @param quotaName {@link QuotaName} object.
+   * @param resourceId resource id of the resource for which check is being done.
    * @return QuotaRecommendation object.
    */
-  protected QuotaRecommendation buildQuotaRecommendation(float usage, QuotaName quotaName) {
+  protected QuotaRecommendation buildQuotaRecommendation(float usage, QuotaName quotaName, String resourceId) {
     QuotaAction quotaAction = QuotaAction.ALLOW;
-    if (requestThrottlingEnabled && usage >= MAX_USAGE_PERCENTAGE_ALLOWED) {
+    if (usage >= MAX_USAGE_PERCENTAGE_ALLOWED) {
+      if (quotaMetrics.perQuotaResourceOutOfQuotaMap.containsKey(resourceId)) {
+        quotaMetrics.perQuotaResourceOutOfQuotaMap.get(resourceId).inc();
+      }
+      if (requestThrottlingEnabled) {
+        quotaAction = QuotaAction.DELAY;
+      }
+    }
```

Comment:
seems like we don't need to set quotaAction in this if statement, the if statement below would do that.